distributed/tests/test_worker.py

```diff
@@ -237,6 +235,20 @@ def f():
         assert w.port > 1024
 
     loop.run_sync(f)
+
+
+def test_worker_waits_for_center_to_come_up(loop):
+    @gen.coroutine
+    def f():
+        w = Worker('127.0.0.1', 8007, ip='127.0.0.1')
+        yield w._start()
+
+    try:
+        loop.run_sync(f, timeout=5)
+    except TimeoutError:
+        pass
```

Comment:
On my machine it takes ~3 seconds for the worker to start waiting. I think anything too close to 3 will not work.

Another way that i can think of is to use Mock for gen and assert that gen.sleep(0.5) is called but that will mean to break this test out into a separate file or make other tests work well with mocks. 

i will keep thinking.